File: qzt-go-server/internal/repository/project/project.go
package project

import (
	"context"

	"gorm.io/gorm"

	projmodel "qzt-go-server/internal/model/project"
	"qzt-go-server/internal/repository"
)

// project.go 项目+任务 repository。

func projDB(ctx context.Context) *gorm.DB { return repository.DBFrom(ctx) }

// ── 项目 ──

type ProjectRepo struct {
	repository.BaseRepo[projmodel.ProjProject]
}

func NewProjectRepo() *ProjectRepo { return &ProjectRepo{} }

func (r *ProjectRepo) PageList(ctx context.Context, page, pageSize int, keyword string, status, priority int8, managerID uint) ([]projmodel.ProjProject, int64, error) {
	var list []projmodel.ProjProject
	q := projDB(ctx).Model(&projmodel.ProjProject{})
	if keyword != "" {
		q = q.Where("name LIKE ? OR project_no LIKE ?", "%"+keyword+"%", "%"+keyword+"%")
	}
	if status > 0 {
		q = q.Where("status = ?", status)
	}
	if priority > 0 {
		q = q.Where("priority = ?", priority)
	}
	if managerID > 0 {
		q = q.Where("manager_id = ?", managerID)
	}
	var total int64
	if err := q.Count(&total).Error; err != nil {
		return nil, 0, err
	}
	err := q.Order("id DESC").Offset((page - 1) * pageSize).Limit(pageSize).Find(&list).Error
	return list, total, err
}

func (r *ProjectRepo) Update(ctx context.Context, m *projmodel.ProjProject) error {
	return r.BaseRepo.Update(ctx, m, "Name", "Description", "CustomerID", "CustomerName", "ContractID", "ManagerID", "MemberIDs", "Status", "Priority", "StartDate", "EndDate", "Progress", "Tags")
}

// ── 任务 ──

type TaskRepo struct {
	repository.BaseRepo[projmodel.ProjTask]
}

func NewTaskRepo() *TaskRepo { return &TaskRepo{} }

func (r *TaskRepo) ListByProject(ctx context.Context, projectID uint) ([]projmodel.ProjTask, error) {
	var list []projmodel.ProjTask
	err := projDB(ctx).Where("project_id = ?", projectID).Order("sort_order ASC, id ASC").Find(&list).Error
	return list, err
}

func (r *TaskRepo) DeleteByProject(ctx context.Context, projectID uint) error {
	return projDB(ctx).Where("project_id = ?", projectID).Delete(&projmodel.ProjTask{}).Error
}

func (r *TaskRepo) Update(ctx context.Context, m *projmodel.ProjTask) error {
	return r.BaseRepo.Update(ctx, m, "Title", "Description", "AssigneeID", "Status", "Priority", "SortOrder", "DueDate", "DoneAt")
}

File: qzt-go-mobile/src/types/project.ts
// 项目管理模块类型

export interface ProjProject {
  id: number
  project_no: string
  name: string
  description: string
  customer_id: number | null
  customer_name: string
  contract_id: number | null
  manager_id: number | null
  member_ids: string
  status: number
  priority: number
  start_date: string | null
  end_date: string | null
  progress: number
  tags: string
  created_at: string
  updated_at: string
}

export interface ProjTask {
  id: number
  project_id: number
  title: string
  description: string
  assignee_id: number | null
  status: number
  priority: number
  sort_order: number
  due_date: string | null
  done_at: string | null
  created_at: string
  updated_at: string
}

export interface ProjectDetail {
  project: ProjProject
  tasks: ProjTask[]
}

export const PROJECT_STATUS: Record<number, { text: string; color: string }> = {
  1: { text: '规划中', color: 'default' },
  2: { text: '进行中', color: 'primary' },
  3: { text: '暂停', color: 'warning' },
  4: { text: '已完成', color: 'success' },
  5: { text: '已取消', color: 'danger' },
}

export const TASK_STATUS: Record<number, { text: string; color: string }> = {
  1: { text: '待办', color: 'default' },
  2: { text: '进行中', color: 'primary' },
  3: { text: '已完成', color: 'success' },
  4: { text: '已取消', color: 'danger' },
}
